Sources/Penny/SOChecker.swift

```diff
@@ -0,0 +1,147 @@
+import Logging
+import DiscordBM
+import Markdown
+import Foundation
+
+actor SOChecker {
+
+    struct Storage: Sendable, Codable {
+        var lastCheckDate: Date? = nil
+    }
+
+    var storage = Storage()
+
+    let soService: any SOService
+    let discordService: DiscordService
+    let logger = Logger(label: "SOChecker")
+
+    init(soService: any SOService, discordService: DiscordService) {
+        self.soService = soService
+        self.discordService = discordService
+    }
+
+    nonisolated func run() {
+        Task { [self] in
+            do {
+                try await self.check()
+                try await Task.sleep(for: .seconds(60 * 5)) /// 5 mins
+            } catch {
+                logger.report("Couldn't check SO questions", error: error)
+                try await Task.sleep(for: .seconds(60 * 5))
+            }
+            self.run()
+        }
```

Comment:
I get what this is doing and why, but it bugs me just the same, if only because you didn't leave yourself any outs by checking for cancellation. Also:
```suggestion
    nonisolated func run() {
        Task { [self] in
            do {
                try await self.check()
            } catch {
                logger.report("Couldn't check SO questions", error: error)
            }
            try await Task.sleep(for: .seconds(60 * 5)) /// 5 mins
            self.run()
        }
```
(And possibly put the catch part in `check()` instead, though I dunno if that really makes more sense or not.)